### PR TITLE
SASS-9998 add balancing charge pages

### PR DESCRIPTION
**Pull request by Matthew-George-Tyler**

### CODE REVIEW

File: app/views/journeys/capitalallowances/balancingCharge/BalancingChargeAmountView.scala.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+@*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (Jake-Raffe) [inline app/views/journeys/capitalallowances/balancingCharge/BalancingChargeAmountView.scala.html:2]: Please update this to 2024

**Comment by Jake-Raffe** [comment]
Please add Specs for the Controllers, Navigator, CYA-Summaries, and Form

File: app/controllers/journeys/capitalallowances/balancingCharge/BalancingChargeAmountController.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BalancingChargeAmountController @Inject() (override val messagesApi: MessagesApi,
+                                                 navigator: CapitalAllowancesNavigator,
+                                                 identify: IdentifierAction,
+                                                 getData: DataRetrievalAction,
+                                                 requireData: DataRequiredAction,
+                                                 service: SelfEmploymentService,
+                                                 formProvider: BalancingChargeAmountFormProvider,
```
Comment (Jake-Raffe) [inline app/controllers/journeys/capitalallowances/balancingCharge/BalancingChargeAmountController.scala:43]: Can the BigDecimalFormProvider be used here instead?

Comment (Matthew-George-Tyler) [reply app/controllers/journeys/capitalallowances/balancingCharge/BalancingChargeAmountController.scala:43]: Probably - I just used the same pattern as previous Controllers which have a custom Form

File: app/views/journeys/capitalallowances/balancingCharge/BalancingChargeView.scala.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            @messages(s"balancingCharge.p1.$userType")
+            @messages(s"balancingCharge.p2")
+            @messages(s"balancingCharge.p3.$userType")
+        </p>
+        <p>
+            @messages(s"balancingCharge.p4.$userType")
+            @messages(s"balancingCharge.p5.$userType")
+        </p>
+        <p>@messages(s"balancingAllowance.p7.read")
+            <a class="govuk-link" target="_blank" href=@messages("balancingAllowance.p7.href") >@messages("balancingCharge.p6.link")</a>
```
[the remark this answers is not in the document]
Comment (Matthew-George-Tyler) [reply app/views/journeys/capitalallowances/balancingCharge/BalancingChargeView.scala.html:49]: This is the same link as used in capital allowances

[the remark this answers is not in the document]
Comment (Jake-Raffe) [reply app/views/journeys/capitalallowances/balancingCharge/BalancingChargeView.scala.html:49]: You're right, sorry, was looking in the wrong place

File: app/views/journeys/capitalallowances/balancingCharge/BalancingChargeAmountView.scala.html (hunk 1 above)
Comment (Matthew-George-Tyler) [reply app/views/journeys/capitalallowances/balancingCharge/BalancingChargeAmountView.scala.html:2]: Done

**Review by Jake-Raffe — CHANGES_REQUESTED** [review]
I don't know if it's because you are force pushing each time but I don't see the changes for
- 'Copyright 2024' in the views
- BigDecimalFormProvider in the controller
- Specs for the Summary viewmodels

**Review by Jake-Raffe — APPROVED** [review]

**Review by kuldeepwadhwa9 — APPROVED** [review]

**Pull request merged**